### DISCUSSION TITLE
WebOS support

### DISCUSSION THREAD

**Discussion by uherman** [jhassan8/crunchyroll-tizen#73] [Q&A]
How is the implementation of WebOS support going?

Is there a project/kanban board or anything where developers looking to contribute can follow the progress and see if there is anything they might do to help? 

I would be happy to contribute, im just not sure where to start since i dont know the current progress of the feature.

**Comment by jhassan8** [comment]
Hi @uherman, I was testing it on an emulator and it worked correctly. currently don't have a TV with webos to test the installation and the app on a device.

**Reply by uherman** [reply]
Interresting, if the code is public i think there are many people who would be happy to test it on a WebOS TV (me included).

Is there a branch to clone somewhere? maybe would be good with some simple steps for how to build the project and get it to run on an emmulator as well, then the step to get it to run on the TV would be quite simple (i think). 😄

**Reply by jhassan8** [reply]
hi @uherman, I uploaded the branch, it is feature/webos,

It has an initial "com.difix.crunchyroll_1.0.0_all.ipk" file to test if it works, you can also use the commands

```
npm run webos-build
npm run webos-package
```

to generate a new ipk.

I understand that on LG there is a tool to install on a rooted TV
https://github.com/webosbrew/dev-manager-desktop

I hope it works, I'm going to try to find someone with an LG TV to try it.

**Reply by Nicolucaspro** [reply]
Do you have any link to the branch? It is to try to test it outright both in the emulator and on the television

**Reply by jhassan8** [reply]
@Nicolucaspro, Did you install the previous application using brew?

the package is https://github.com/jhassan8/crunchyroll-tizen/blob/feature/webos/com.difix.crunchyroll_1.0.0_all.ipk
I think they are installed on the TV with [webosbrew/dev-manager-desktop](https://github.com/webosbrew/dev-manager-desktop),
The emulator can be downloaded from https://webostv.developer.lge.com/develop/tools/emulator-installation, if you have any questions, let me know.

**Reply by Nicolucaspro** [reply]
Thanks for the information, I will let you know how it works on my television.

**Reply by Nicolucaspro** [reply]
@jhassan8 I hope to be of some use to the project of these televisions by reporting bugs and other things.

**Comment by Nicolucaspro** [comment]
I am very excited, this is what it looks like inside the application, thank you very much for having created it, I just need to make it public. I will inform you of any news.
![IMG_20240113_171412](https://github.com/jhassan8/crunchyroll-tizen/assets/156543495/a702c78d-c7b8-4f9c-b4d4-843a51513c62)

**Reply by Nicolucaspro** [reply]
It receives you as the app would receive you on a Samsung television with the news announcements

**Reply by jhassan8** [reply]
Great that it works, is that photo working on the TV?

**Reply by Nicolucaspro** [reply]
@jhassan8 This photo comes from the television that I use daily in my living room. I have been testing the application and noticed a couple of bugs.

**Reply by Nicolucaspro** [reply]
@jhassan8 I sent an email to the one you have on your Github profile page about it.

**Comment by KTheMan** [comment]
> How is the implementation of WebOS support going?
> 
> Is there a project/kanban board or anything where developers looking to contribute can follow the progress and see if there is anything they might do to help? 
> 
> I would be happy to contribute, im just not sure where to start since i dont know the current progress of the feature.


Worked great for me! There are some great quality of life features in your implementation @jhassan8. I'm blown away that we have a skip intro feature and autoplay...

After connecting over SSH, it's as easy as 2 lines:


```
wget https://github.com/jhassan8/crunchyroll-tizen/releases/download/v1.1.5/com.difix.crunchyroll_1.0.0_all.ipk
luna-send-pub -i 'luna://com.webos.appInstallService/dev/install' '{"id":"com.ares.defaultName","ipkUrl":"/home/root/com.difix.crunchyroll_1.0.0_all.ipk","subscribe":true}'
```


Edit: I guess code blocks don't work in discussions... Oops. Leaving anyway

**Reply by KTheMan** [reply]
Would also be great to have this on the Homebrew Store 
https://repo.webosbrew.org/submit
